components/cop_codegen/src/rpn_function.rs

```diff
@@ -255,6 +256,25 @@ fn parse_arg_type(arg: &FnArg) -> Result<TypePath> {
     Ok(eval_type.clone())
 }
 
+fn check_attr(meta: &[Ident], item_fn: &ItemFn) -> Result<()> {
+    if item_fn.decl.inputs.len() < meta.len() {
+        return Err(meta[0]
+            .span()
+            .unwrap()
+            .error("The attrs of `[rpn_fn]` are more than the params of the function"));
+    }
+    for m in meta {
+        if m != "ctx" && m != "payload" {
+            return Err(meta[0].span().unwrap().error(
+                "The attr of `[rpn_fn]` can only be `ctx` or `payload`,\
```

Comment:
Missing space after `,`. `#[rpn_fn]` may be better
I think this sentence can end with a period. Then, "Make sure..."
(In fact, I think it's ok to even remove the lattter setence...)